### PR TITLE
Two bugs; one commit

### PR DESCRIPTION
**Pull request by bmeike**
This is fixes for:
CBL-2389 | Confirm LiteCore request for close, before connection is opened.
CBL-2395 | Kotlin FullTextIndexConfiguration?.create extension function does not support multiple expressions

### CODE REVIEW

File: common/main/java/com/couchbase/lite/FullTextIndexConfiguration.java [hunk 1]
```diff
@@ -31,6 +32,8 @@ public FullTextIndexConfiguration(@NonNull String... expressions) {
         super(IndexType.FULL_TEXT, Arrays.asList(expressions));
     }
 
+    FullTextIndexConfiguration(@NonNull List<String> expressions) { super(IndexType.VALUE, expressions); }
```
Comment (pasin) [inline common/main/java/com/couchbase/lite/FullTextIndexConfiguration.java:35]: Should it be IndexType.FULL_TEXT?

**Review by pasin — CHANGES_REQUESTED** [review]

File: common/main/java/com/couchbase/lite/FullTextIndexConfiguration.java (hunk 1 above)
Comment (bmeike) [reply common/main/java/com/couchbase/lite/FullTextIndexConfiguration.java:35]: OMG... OMG... What a catch!  Thanks!

Comment (bmeike) [reply common/main/java/com/couchbase/lite/FullTextIndexConfiguration.java]: done

**Comment by bmeike** [comment]
Please note: A complete fix for the root problem would, as noted in the related CBSE, require a timeout on silent connections.  Since that scenario is quite unlikely, and introducing a timer into sensitive code is a significant change, I am not going to do it.  The fix simply addresses the case in which the Core closes a connection that is not yet open.

**Review by pasin — APPROVED** [review]

**Comment by bmeike** [comment]
Also note that it is quite difficult to write a dev test to confirm that this fix works: I have not added one.

I will have to set up a test network to try this out, probably using Jim's net-breaking proxy.  Will confirm results as soon as I have them.

It should be relatively easy for QE to test this scenario.  I hope that they will add such a test.

**Comment by pasin** [comment]
I think OKHTTP might already have connection timeout.

File: common/main/java/com/couchbase/lite/internal/replicator/AbstractCBLWebSocket.java [hunk 2]
```diff
@@ -175,7 +184,10 @@ public void onMessage(@NonNull WebSocket webSocket, @NonNull ByteString bytes) {
         public void onClosing(@NonNull WebSocket webSocket, int code, @NonNull String reason) {
             Log.d(TAG, "%s#OkHTTP closing: %s", AbstractCBLWebSocket.this, reason);
             synchronized (getPeerLock()) {
-                if (!state.setState(State.CLOSE_REQUESTED)) { return; }
+                if (!state.setState(State.CLOSE_REQUESTED))  {
+                    if (state.assertState(State.CLOSED)) { webSocket.cancel(); }
```
Comment (pasin) [inline common/main/java/com/couchbase/lite/internal/replicator/AbstractCBLWebSocket.java:188]: I couldn't think about the case that the state is already CLOSED while OKHTTP is closing. However it doesn't hurt to put the logic here for precaution.

**Pull request closed without merging**